test/integration/test_fetchcontent_source_dir.rb

```diff
@@ -0,0 +1,43 @@
+require_relative './lib'
+
+# Tests FetchContent overriding with CPM
+
+class FetchContentSourceDir < IntegrationTest
+  def setup
+    @cache_dir = File.join(cur_test_dir, 'cpmcache')
+    ENV['CPM_SOURCE_CACHE'] = @cache_dir
+  end
+
+  def test_add_dependency_cpm_and_fetchcontent
```

Comment:
The function's name should also be changed

```suggestion
  def test_fetchcontent_source_dir
```

Though perhaps a more descriptive name for this and the test file would be better? Maybe something like "test_remove_source_dir" / "RemoveSourceDir"?